packages/core/admin/admin/src/content-manager/components/BlocksInput/utils/enterKey.ts

```diff
@@ -0,0 +1,48 @@
+import { type Text, Editor, Node, Transforms } from 'slate';
+
+const isText = (node: unknown): node is Text => {
+  return Node.isNode(node) && !Editor.isEditor(node) && node.type === 'text';
+};
+
+/**
+ * Inserts a line break the first time the user presses enter, and exits the node the second time.
+ */
+const pressEnterTwiceToExit = (editor: Editor) => {
+  /**
+   * To determine if we should break out of the node, check 2 things:
+   * 1. If the cursor is at the end of the node
+   * 2. If the last line of the node is empty
+   */
+  const nodeEntry = Editor.above(editor, {
+    match: (node) => !Editor.isEditor(node) && !['link', 'text'].includes(node.type),
+  });
+  if (!nodeEntry || !editor.selection) {
+    return;
+  }
+  const [node, nodePath] = nodeEntry;
+  const isNodeEnd = Editor.isEnd(editor, editor.selection.anchor, nodePath);
+  const lastTextNode = node.children.at(-1);
+  const isEmptyLine = isText(lastTextNode) && lastTextNode.text.endsWith('\n');
+
+  if (isNodeEnd && isEmptyLine) {
+    // Remove the last line break
+    Transforms.delete(editor, { distance: 1, unit: 'character', reverse: true });
+    // Break out of the node by creating a new paragraph
+    Transforms.insertNodes(editor, {
+      type: 'paragraph',
+      children: [{ type: 'text', text: '' }],
+    });
+  } else {
+    // Otherwise insert a new line within the node
+    Transforms.insertText(editor, '\n');
+
+    // If there's nothing after the cursor, disable modifiers
+    if (isNodeEnd) {
+      for (const modifier of ['bold', 'italic', 'underline', 'strikethrough', 'code']) {
+        Editor.removeMark(editor, modifier);
+      }
```

Comment:
Because I saw something recently saying for const had slightly better perf, since you're not creating functions.

But I just checked and we don't use it in strapi, so I'll stick to forEach for consistency

Reply:
There used to be an eslint rule on the frontend that wouldn't allow it. I don't really care though if you prefer this that's cool.